Split `FieldOfView` into subclasses

teutoburg opened the pull request:
The `FieldOfView` used to produce either 1D (unused before MOS), 2D (imaging) or 3D (spectroscopy) outputs. The distinction between 2D and 3D was actually made well before in the `FOVManager`. Meaning by the time the `FieldOfView` object is created, it is already known which type of output will be produced. So it is not really needed at all to retain all three options all the way through the optical train.

In my opinion, splitting this into subclasses makes it quite a bit better structured to look at and understand what's happening in which case, and what attributes and methods are shared (i.e. belong to the base class). I used the `__new__()` "trick" to still enable creation via `fov = FieldOfView()` with the `hdu_type` argument as a distinction. I might change this later on to only directly instantiate the three subclasses, that depends somewhat on how the MOS stuff is implemented, because it will be the first thing to actually use the 1D variant (outside of unit tests).

I think refactoring the various `_make_xy` methods into something more consistent (think BUNIT on 2+1D sources) will be easier to navigate with this split.

hugobuddel reviewed (commented):
Great to have these derived classes!

I don't understand why the `__new__()` and `hdu_type` is necessary, see the other comments. It seems unnecessarily complex at first sight.

teutoburg commented:
> I don't understand why the `__new__()` and `hdu_type` is necessary, see the other comments. It seems unnecessarily complex at first sight.

Yeah, you're right. This is silly. I think my original though process was something like this: I looked at how I implemented the 2D/3D "dispatch" in the source fields and thought that neat, perhaps I could do something similar here (since it's also a 1D/2D/3D task). But then halfway through I realized that I couldn't determine the dimensionality from anything else, so I added the `hdu_type` parameter. I didn't like that it ended up in the init's kwargs (see other comment), but accepted it.

Now in hindsight it seems I didn't see the forest for the trees. I also found that the FOV is instantiated only in very few places in the code, and probably never "from the outside". So replacing the part where `hdu_type` is selected with the correct subclass is certainly the better way to got. I'll change that.

teutoburg commented on scopesim/optics/image_plane.py:
```diff
@@ -106,6 +106,10 @@ def add(self, hdus_or_tables, sub_pixel=None, spline_order=None,
 
+        .. versionchanged:: PLACEHOLDER_NEXT_RELEASE_VERSION
+
+           Here's a version changed message.
```
I cannot figure out what my thought was behind adding this here, so I'm removing it...
